storage/store.go

```diff
@@ -989,11 +989,14 @@ func (s *Store) MergeRange(subsumingRng *Replica, updatedEndKey proto.Key, subsu
 
 	// Remove and destroy the subsumed range. Note that we are on the
 	// processRaft goroutine so we can call removeReplicaImpl directly.
-	if err = s.removeReplicaImpl(subsumedRng); err != nil {
+	if err := s.removeReplicaImpl(subsumedRng); err != nil {
 		return util.Errorf("cannot remove range %s", err)
 	}
 
-	// TODO(bram): The removed range needs to have all of its metadata removed.
+	localRangeKeyPrefix := keys.MakeLocalRangeKeyPrefix(subsumedDesc.RangeID)
```

Comment:
We also need to delete `keys.RangeDescriptorKey()` for the subsumed range (but *not* the rest of the `keys.MakeRangeKey` prefix, since the transaction records etc are adopted by the subsuming range)